test/base.test.js

```diff
@@ -65,16 +98,27 @@ test('should support multiple secrets', async (t) => {
   const fastify = await buildFastify(handler, options, plugin)
   t.teardown(() => fastify.close())
 
-  const response = await fastify.inject({
+  const sessionIdSignedWOldSecret = 'aYb4uTIhdBXCfk_ylik4QN6-u26K0u0e.InCp31AuDa7DX%2F8rGBz8RMFiCpmUtjcF%2BS7Aco7tur8'
+  const response1 = await fastify.inject({
     url: '/',
     headers: {
       'x-forwarded-proto': 'https',
-      cookie: 'sessionId=aYb4uTIhdBXCfk_ylik4QN6-u26K0u0e.eiVu2YbrcqbTUYTYaANks%2Fjn%2Bjta7QgpsxLO%2BOLN%2F4U; Path=/; HttpOnly; Secure'
+      cookie: `sessionId=${sessionIdSignedWOldSecret}; Path=/; HttpOnly; Secure`
     }
   })
+  t.equal(response1.statusCode, 200)
+  t.equal(response1.headers['set-cookie'].includes(sessionId), true)
 
-  t.equal(response.statusCode, 200)
-  t.equal(response.headers['set-cookie'].includes('aYb4uTIhdBXCfk_ylik4QN6-u26K0u0e'), false)
```

Comment:
As explained in:
https://github.com/fastify/session/pull/101#discussion_r942776850

This was a bad test. This assertion is testing that there is a new session. This is wrong, the existing session should remain -- the only reason there's a new session is because `sessionId` was not stored within the session.

After our changes, we see different behavior.